FlagsmithClient/src/main/java/com/flagsmith/internal/FlagsmithApi.kt

```diff
@@ -9,11 +9,11 @@ import com.github.kittinunf.fuel.core.Parameters
 import com.github.kittinunf.fuel.util.FuelRouting
 import com.google.gson.Gson
 
-sealed class FlagsmithApi: FuelRouting {
-    class getIdentityFlagsAndTraits(val identity: String): FlagsmithApi()
-    class getFlags(): FlagsmithApi()
-    class setTrait(val trait: Trait, val identity: String): FlagsmithApi()
-    class postAnalytics(val eventMap: Map<String, Int?>): FlagsmithApi()
+sealed class FlagsmithApi : FuelRouting {
+    class GetIdentityFlagsAndTraits(val identity: String) : FlagsmithApi()
+    object GetFlags : FlagsmithApi()
```

Comment:
Just checking the reasoning behind changing the type here to `object` vs the rest that still seem to have `class` ?

Reply:
It's common practice in kotlin to tend towards the simplest type of data structure. A `class` always has to be instantiated but a `object` behaves similarly to a singleton. It is a single reference to some collection of data. They are commonly used in `sealed interfaces/classes` as they don't contain a lot of data and are just meant as a simple representation.

More specifically in this case, the `sealed class` is acting in a similar way to an `enum` but it can some `enum` values can take parameters. The `object` is used for values that don't take parameters.

Reply:
Got it, thanks for explaining!